### PR TITLE
Feature/data management

### LINKED ISSUE #465: Phase 5: Data Management Tools
The goal of this phase is to implement the **Data Management** tool group. These tools handle the full data lifecycle — loading, inspecting, splitting, transforming, and persisting time series data — providing LLM clients with a structured, handle-based interface for working with data *before* any model is fitted.

---

### 1. `inspect_data`
Returns rich metadata about a loaded data handle, including its sktime machine type, scientific type, shape, frequency, cutoff point, missing value counts, a preview of the data, and summary statistics.

**Arguments:**

- `data_handle` (`str`, required): Handle ID from `load_data_source`.

**Returns:**

- `dict`: `{"mtype": str, "scitype": str, "shape": list[int], "columns": list[str], "dtypes": dict, "index_names": list[str], "freq": str|null, "cutoff": str|null, "n_missing": int, "head": dict, "summary_stats": dict}`

---

### 2. `split_data`
Performs a temporal train/test split on a data handle, registering both halves as new handles. Exactly one of `test_size` or `fh` must be provided.

**Arguments:**

- `data_handle` (`str`, required): Handle ID from `load_data_source`.
- `test_size` (`float`, optional): Fraction of the data to hold out for testing (0.0–1.0). Mutually exclusive with `fh`.
- `fh` (`list[int]` | `int`, optional): Forecast horizon — number of final time steps to reserve as the test set. Mutually exclusive with `test_size`.

**Returns:**

- `dict`: `{"train_handle": str, "test_handle": str, "cutoff": str, "train_size": int, "test_size": int}`

---

### 3. `transform_data`
Unified data transformation tool with two modes:

- **`action="format"`** — Auto-fix common time series issues: infer and set frequency, remove duplicate timestamps, and fill missing values using forward/backward fill. This subsumes the existing `format_time_series` tool.
- **`action="convert"`** — Convert the data to a different sktime machine type (mtype) using `sktime.datatypes.convert_to()`. Requires the `to_mtype` argument.

**Arguments:**

- `data_handle` (`str`, required): Handle ID from `load_data_source`.
- `action` (`str`, optional, default=`"format"`): `"format"` or `"convert"`.
- `auto_infer_freq` (`bool`, optional, default=`true`): (format only) Infer and set frequency.
- `fill_missing` (`bool`, optional, default=`true`): (format only) Fill missing values.
- `remove_duplicates` (`bool`, optional, default=`true`): (format only) Remove duplicate timestamps.
- `to_mtype` (`str`, optional): (convert only) Target mtype, e.g. `"pd.DataFrame"`, `"pd.Series"`, `"np.ndarray"`.

**Returns:**

- `dict`: `{"data_handle": str, "changes_applied": list[str]}`

---

### 4. `save_data`
Persists the data behind a handle to a local file. Supports CSV, Parquet, and JSON output formats. Combines the target series (`y`) and exogenous features (`X`) into a single export.

**Arguments:**

- `data_handle` (`str`, required): Handle ID from `load_data_source` / `split_data` / `transform_data`.
- `path` (`str`, required): Destination file path.
- `format` (`str`, optional, default=`"csv"`): Output format — `"csv"`, `"parquet"`, or `"json"`.

**Returns:**

- `dict`: `{"saved_path": str, "format": str, "rows": int}`

---

## Implementation Notes

- **`inspect_data`**: Use `sktime.datatypes.scitype()` for scitype detection, `sktime.datatypes.get_cutoff()` for cutoff extraction, and standard pandas introspection (`.describe()`, `.head()`, `.isna()`) for summary stats and missing value counts. Include fallback heuristics when sktime functions are unavailable.
- **`split_data`**: Use `sktime.split.temporal_train_test_split()` when available, falling back to plain pandas `.iloc` slicing. Register both train and test portions as new data handles via `executor._register_data_handle()`, inheriting metadata from the parent handle.
- **`transform_data`**: The `"format"` action delegates to `executor.format_data_handle()` (the same backend as the existing `format_time_series` tool). The `"convert"` action uses `sktime.datatypes.convert_to()` and registers the result as a new handle.
- **`save_data`**: Use `pathlib.Path` for all filesystem operations (per ruff PTH rules). Dispatch to the appropriate pandas writer method (`to_csv`, `to_parquet`, `to_json`) based on the `format` argument.

---

## Acceptance Criteria

- [ ] Implement the `inspect_data`, `split_data`, `transform_data`, and `save_data` tools as defined above.
- [ ] All tools operate on data handles registered via `load_data_source` — no raw data is passed through the MCP interface.
- [ ] Define precise JSON schemas in `server.py` for the tool arguments and expected outputs (via MCP `Tool` definitions with `inputSchema`).
- [ ] Wire all 4 tools into the `call_tool()` dispatcher in `server.py`.
- [ ] Export all 4 tools from `tools/__init__.py`.
- [ ] Add unit tests covering:
  - `inspect_data`: all metadata fields present, unknown handle error.
  - `split_data`: split by `test_size`, split by `fh`, mutual exclusivity validation, invalid `test_size`, unknown handle.
  - `transform_data`: format action, convert requires `to_mtype`, invalid action, unknown handle.
  - `save_data`: CSV export, JSON export, unsupported format, unknown handle.
- [ ] Ensure all existing tests continue to pass (`make check`).
- [ ] The existing `format_time_series` tool remains available for backward compatibility.

---

## Relationship to Existing Tools

| Existing Tool | Status | Relationship |
|---------------|--------|-------------|
| `load_data_source` | **Kept** | Provides the data handles that all new tools operate on |
| `load_data_source_async` | **Kept** | Async variant of load |
| `list_available_data` | **Kept** | Lists available demo datasets and active handles |
| `format_time_series` | **Kept (backward compat)** | `transform_data(action="format")` wraps the same logic |
| `release_data_handle` | **Kept** | Frees handles created by any data tool |

---

## sktime Coverage

| sktime Subsystem | Covered By |
|-----------------|-----------|
| 33 demo datasets | `load_data_source(source=...)` |
| `mtype()`, `check_is_mtype()` | `inspect_data` |
| `get_cutoff()` | `inspect_data` |
| Scitype detection | `inspect_data` |
| `temporal_train_test_split()` | `split_data` |
| `convert_to()` | `transform_data(action="convert")` |
| Frequency inference, fill NaN, dedup | `transform_data(action="format")` |
| File export (CSV/Parquet/JSON) | `save_data` |

### PR DESCRIPTION
**Pull request by Shashankss1205**
# PR: Add Data Management Tools

#### Reference Issues/PRs

Fixes #465

**Independence:** This PR is rebased on latest `main` and does **not** depend on #463 or any other open PR. It can be merged on its own. (Previously bundled #463 commits; those have been removed so this PR contains only data-management changes.)

#### What does this implement/fix? Explain your changes.

Adds 4 new MCP tools for the Data Management group:

| # | New Tool | Purpose | sktime Coverage |
|:-:|----------|---------|-----------------|
| 4 | **`inspect_data`** | Rich metadata inspection of loaded data handles | `mtype()`, `check_is_mtype()`, `get_cutoff()`, scitype detection |
| 5 | **`split_data`** | Temporal train/test splitting with `test_size` (fraction) or `fh` (horizon count) | `temporal_train_test_split()` |
| 6 | **`transform_data`** | `action="format"` (auto-fix freq/dupes/NaN) or `action="convert"` (mtype conversion) | `convert_to()`, frequency inference, fill NaN, dedup |
| 7 | **`save_data`** | Persist data handles to CSV, Parquet, or JSON files | File export |

**Files created:**
- `src/sktime_mcp/tools/inspect_data.py`
- `src/sktime_mcp/tools/split_data.py`
- `src/sktime_mcp/tools/transform_data.py`
- `src/sktime_mcp/tools/save_data.py`
- `tests/test_data_management.py`

**Files modified:**
- `src/sktime_mcp/server.py` — tool schemas + dispatcher
- `src/sktime_mcp/tools/__init__.py` — exports
- `README.md` — tool documentation

The existing `format_time_series` tool is preserved for backward compatibility; `transform_data(action="format")` delegates to the same executor logic.

#### Does your contribution introduce a new dependency? If yes, which one?

No. Uses existing dependencies (`pandas`, `sktime`).

#### What should a reviewer concentrate their feedback on?

- [x] `inspect_data` — mtype/scitype detection fallback logic
- [x] `split_data` — temporal splitting and handle registration
- [x] `transform_data` — `convert_to()` edge cases
- [x] `save_data` — format dispatch and path handling

#### Any other comments

Rebased on latest `main` (3 commits, +1282 / −31). **197 tests** pass under `make check`.

##### For all contributions
- [x] I've added unit tests and made sure they pass locally (`make check`).
- [ ] I've added the tool to the online documentation in `docs/source/`.

### CODE REVIEW

**Review by fkiraly — CHANGES_REQUESTED** [review]
Siimilar to the other PR. Why do we reinvent the wheel and duplicate time series handling logic in `sktime`?

* I think many of the data metadata fields can be obtained via `check_is_mtype` or similar. We should assume only a few possible data types for the start.
* The splitting seems to duplicate code in `sktime.split`.

I get that many of the dulications seems to come from extremely defensive try/except code - but why that? Are the sktime methods not reliably enough?

Your standard pattern seems to be in this PR:

```python
def _my_method_xyz(a, b, c):
    try:
         from sktime.abc import xyz
         return xyz(a, b, c)
    except:
        lots of duplicative code that copies in ad-hoc manner xyz
```
